torch-neuronx/training/fsdp_gpt2/run_clm_slurm_mics.sh

```diff
@@ -0,0 +1,127 @@
+    export NEURON_CC_FLAGS="--retry_failed_compilation --internal-dram-page-size=2048 --enable-experimental-spmd --model-type transformer --internal-max-instruction-limit=10000000 --enable-experimental-O1 --internal-build-with-users"
+fi
+
+#export PARTITION_WORLD_SIZE=$NUM_NEURONCORES #TODO: remove with torchneuronx MICS
+export DISABLE_NUMERIC_CC_TOKEN=1
+export NEURON_RT_HIERARCHICAL_CC=1
+
+# Mics
+export NEURON_MICS_PARTITION_GROUP_SIZE=$NUM_NEURONCORES
+
+#TODO : Change defaults for these to something more useful (eg the values in this script !)
+#TODO : FAL automatically inserts --internal-dram-page-size if RT env var is set
+#This needs to match compiler option --internal-dram-page-size if it is set
+export NEURON_RT_ONE_TMPBUF_PAGE_SIZE_MB=2048 #Need to match
+export NEURON_RT_EXEC_TIMEOUT=600
+export TF_NUM_INTEROP_THREADS=8192
+
+GRAD_ACCUM_STEP=1
+BATCH_SIZE=1
+MODEL_CONFIG="config_1p5B_gpt2.json"
+#MODEL_CONFIG="config_125M_gpt2_small.json"
+MODEL_SIZE=$(echo $MODEL_CONFIG | grep -m 1 -Eo '[0-9MBp]+' | head -n1 | tr -d '\n')
+DATASET_CONFIG=$2
+
+if [ $GRAD_ACCUM_STEP -gt 1 ]; then
+    echo "need to uncomment accelerator.py code to run"
+    ./uncomment_gradaccum.sh
+fi
+
+MAX_STEPS=100000
+LOG_FILE_NAME="run_log_hf_gpt2_param_"$MODEL_SIZE"_nodes"$WORLD_SIZE_JOB"_grad_accum"$GRAD_ACCUM_STEP"_bs"$BATCH_SIZE_$(date +"%m-%d-%Y")_$(date +"%H:%M:%S")
+if [[ "$NEURON_EXTRACT_GRAPHS_ONLY" == "1" ]]; then
+    EPOCHS=1
+    MAX_STEPS=10
+    LOG_FILE_NAME="compile_log_hf_gpt2_param_"$MODEL_SIZE"_grad_accum"$GRAD_ACCUM_STEP"_bs"$BATCH_SIZE_$(date +"%m-%d-%Y")_$(date +"%H:%M:%S")
+fi
+
+
+torchrun $DISTRIBUTED_ARGS run_clm_no_trainer.py \
+    --model_name_or_path gpt2 \
+    --dataset_name wikitext \
+    --dataset_config_name $DATASET_CONFIG \
+    --config_name $MODEL_CONFIG \
+    --per_device_train_batch_size $BATCH_SIZE \
+    --gradient_accumulation_steps $GRAD_ACCUM_STEP \
+    --max_train_steps $MAX_STEPS \
+    --weight_decay 0.01 \
+    --learning_rate 0.00015 \
+    --lr_scheduler_type cosine \
+    --use_fsdp \
+    --gradient_checkpointing \
+    --seed 1234 \
+    --use_mics \
+    --num_warmup_steps 75 \
+    --use_grad_clipping \
+    --validation_split_percentage 0 \
+    --output_dir test-clm \
+    |& tee $LOG_FILE_NAME &
+wait %1
+
+ret_val=$?
+if [ $ret_val -eq 0 ]; then
+    success=1
+else
+    success=0
+fi
+
+dump_to_s3_update_json_scr=../../../dump_to_s3_update_test_json.sh
```

Comment:
this is for pipeline, in this sample repo, I don't think we have `dump....json.sh`